src/engine/Result.cpp

```diff
@@ -279,14 +279,21 @@ void Result::cacheDuringConsumption(
           const IdTableVocabPair& newTablePair) {
         bool doBothFitInCache = fitInCache(aggregate, newTablePair);
         if (doBothFitInCache) {
-          if (aggregate.has_value()) {
-            auto& value = aggregate.value();
-            value.idTable_.insertAtEnd(newTablePair.idTable_);
-            value.localVocab_.mergeWith(
-                std::span{&newTablePair.localVocab_, 1});
-          } else {
-            aggregate.emplace(newTablePair.idTable_.clone(),
-                              newTablePair.localVocab_.clone());
+          try {
+            if (aggregate.has_value()) {
+              auto& value = aggregate.value();
+              value.idTable_.insertAtEnd(newTablePair.idTable_);
+              value.localVocab_.mergeWith(
+                  std::span{&newTablePair.localVocab_, 1});
+            } else {
+              aggregate.emplace(newTablePair.idTable_.clone(),
+                                newTablePair.localVocab_.clone());
+            }
+          } catch (...) {
```

Comment:
In this case here sonarcloud is right: You can catch a specific AllocationExceedsMemoryWhateverItIsCalledException.

Reply:
Thought about that but the idea was to not let any other exception like `std::bad_alloc` for example interrupt the operation too